crates/uv-resolver/src/requires_python.rs

```diff
@@ -362,8 +362,9 @@ impl RequiresPython {
                     let Some(minor) = python_tag
                         .strip_prefix("cp3")
                         .or_else(|| python_tag.strip_prefix("pp3"))
-                        .or_else(|| python_tag.strip_prefix("py3"))
                     else {
+                        // We also return true for bounds such as `py36`, where the Python version
+                        // is a lower bound (redundant with `requires-python`)
```

Comment:
What if the version is greater than our own upper-bound on Python? E.g., `py310` but the user has `== 3.8.*`.

Reply:
I guess we don't handle this at all here yet (upper bounds).